apps/ae_socket_connector/lib/client_runner.ex

```diff
@@ -103,35 +107,85 @@ defmodule ClientRunner do
     end
   end
 
-  def process_sign_request(message, to_sign, state) do
-    try do
-      case elem(message, 0) do
-        # TODO how do we descide if we should sign?
-        :sign_approve ->
-          signed = SessionHolder.sign_message(state.pid_session_holder, to_sign)
-          fun = fn pid -> SocketConnector.send_signed_message(pid, elem(message, 2), signed) end
-          SessionHolder.run_action(state.pid_session_holder, fun)
+  def process_sign_request(message, to_sign, pid_session_holder, mode \\ %{sign: {:default}})
 
-        _ ->
-          :ok
-      end
-    rescue
-      _ignore -> :ok
+  def process_sign_request({}, _to_sign, _pid_session_holder, _mode) do
+    Logger.debug("Empty request")
+  end
+
+  def process_sign_request(message, to_sign, pid_session_holder, %{sign: sign_info}) do
+    case elem(message, 0) do
+      :sign_approve ->
+        case sign_info do
+          {:default} ->
+            signed = SessionHolder.sign_message(pid_session_holder, to_sign)
+            fun = fn pid -> SocketConnector.send_signed_message(pid, elem(message, 2), signed) end
```

Comment:
Please don't use `elem/2` so much. Match the values up top somewhere and use them directly here. Same in the rest of the code.